### PR TITLE
newdevice: initial proxmox detection

### PR DESCRIPTION
**Pull request by tomarch**
Dectect PROXMOX host

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: includes/definitions/proxmox.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+discovery_modules:
+    applications: 1
+    bgp-peers: 0
+    stp: 0
+    vmware-vminfo: 0
+    libvirt-vminfo: 0
+processor_stacked: 1
+discovery:
+    - sysDescr_regex:
+        - '/PVE/'
```
Comment (murrant) [inline includes/definitions/proxmox.yaml]: This doesn't need to be a regex.  You are just searching for PVE anywhere in the sysDescr.
```
 - sysDescr: PVE
```

**Comment by murrant** [comment]
The os icon should be squarish.  Move the current svg you have to the logos directory and extract the left symbol for this os icon.

File: includes/definitions/proxmox.yaml [hunk 2]
```diff
@@ -21,5 +21,7 @@ discovery_modules:
     libvirt-vminfo: 0
 processor_stacked: 1
 discovery:
-    - sysDescr_regex:
-        - '/PVE/'
+    - sysObjectId:
```
Comment (murrant) [inline includes/definitions/proxmox.yaml]: This is wrong, it means match either sysObjectID or sysDescr.   You need to have them in the same array (-).

File: includes/definitions/proxmox.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    applications: 1
+    bgp-peers: 0
+    stp: 0
+    vmware-vminfo: 0
+    libvirt-vminfo: 0
+processor_stacked: 1
+discovery:
+    - sysObjectId:
+            - .1.3.6.1.4.1.8072.3.2.10
+    - sysDescr:
```
Comment (murrant) [inline includes/definitions/proxmox.yaml]: This is wrong, it means match either sysObjectID or sysDescr. You need to have them in the same array (-).
Reply…

Comment (tomarch) [reply includes/definitions/proxmox.yaml]: Sorry. Should be fixed now.

File: includes/definitions/proxmox.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    stp: 0
+    vmware-vminfo: 0
+    libvirt-vminfo: 0
+processor_stacked: 1
+discovery:
+    -
+      sysObjectId:
+            - .1.3.6.1.4.1.8072.3.2.10
+      sysDescr:
+            - PVE
```
Comment (laf) [inline includes/definitions/proxmox.yaml]: I think to be 100% safe here it might be better to do:

```
sysDescr_regex:
    - '/ PVE /'
```

Comment (murrant) [reply includes/definitions/proxmox.yaml]: @laf just a note, this didn't need to be a regex `- ' PVE '` works too

Comment (laf) [reply includes/definitions/proxmox.yaml]: Good point

**Pull request merged**